### PR TITLE
Videos without instructions

### PR DESCRIPTION
**Pull request by islemaster**
Right now, if you set up a level in levelbuilder with a video but leave the instructions box empty, the video will not auto-play.  This isn't usually a problem (most levels have instructions) but it can be confusing while building a level.

With this change, levels with no instructions can still autoplay videos.

Shared levels or levels with "Skip instruction popup" checked will still skip both the autoplay video and the instructions popup.

In testing this change I had several UI tests fail related to closing instructions so I applied the more stable "close the dialog" step in those cases to improve the test stability.

### CODE REVIEW

File: shared/js/initApp.js [hunk 1]
```diff
@@ -61,17 +61,22 @@ var baseOptions = {
...
-    var hasInstructions = appOptions.level.instructions || appOptions.level.aniGifURL;
-    if (!hasInstructions || this.share || appOptions.level.skipInstructionsPopup) {
+    // Always skip all pre-level popups on share levels or when configured thus
+    if (this.share || appOptions.level.skipInstructionsPopup) {
       return;
     }
 
-    if (appOptions.autoplayVideo) {
+    var hasVideo = !!appOptions.autoplayVideo;
+    var hasInstructions = appOptions.level.instructions || appOptions.level.aniGifURL;
```
Comment (Bjvanminnen) [inline shared/js/initApp.js]: Any reasoning for using !! on the first but not the second? Seems like with our without should be valid in both cases, but might want to be consistent.

**Comment by Bjvanminnen** [comment]
lgtm. what are the risks that you're worried about?

**Comment by islemaster** [comment]
I suppose the only undesirable behavior for this change would occur on levels that:

1. Had a video specified in levelbuilder
2. Had no instructions and no ani-Gif

In this case, those levels would now have an autoplay video where they did not before.  I'll search for levels that would meet those conditions.

**Comment by islemaster** [comment]
Okay, I didn't find any `.level` files that have videos but no instructions, so we should be good.

**Pull request merged**